pkg/cli/admin/inspect/namespace.go

```diff
@@ -92,3 +101,65 @@ func (o *InspectOptions) gatherNamespaceData(baseDir, namespace string) error {
 	}
 	return nil
 }
+
+// podEligibleForCollection aims to filter out the daemonset pods running on
+// data planes if node count is greater than a threshold,
+// in order to save significant storage size and reduces the collection duration.
+// If these pods need to be collected, recommended path is
+// to use node-selector flag (i.e. --node-selector='node-role.kubernetes.io/worker').
+func (o *InspectOptions) podEligibleForCollection(pod *corev1.Pod) (bool, error) {
+	isDaemonSet := false
+	if len(pod.OwnerReferences) > 0 {
+		for _, OwnerRef := range pod.OwnerReferences {
+			if OwnerRef.Kind == "DaemonSet" {
+				isDaemonSet = true
+				break
+			}
+		}
+	}
+	if !isDaemonSet {
+		return true, nil
+	}
+
+	nodes, err := o.kubeClient.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return true, err
+	}
+	const daemonsetCollectionMaxNodeSize = 20
+	if nodes.Items == nil || len(nodes.Items) < daemonsetCollectionMaxNodeSize {
```

Comment:
Given this is mainly for the data nodes what about to extract the number of control plane nodes from len(nodes.Items) first and then check for the limit?

Reply:
20 is for total number of nodes in the cluster regardless of data plane or control plane. Do you suggest using a threshold based on the number of data plane nodes?. It makes sense. Do you have any threshold suggestion?. What about 15?

Reply:
```
$ echo $((10 + $RANDOM % 10))
18
```

Reply:
Thank you for this, friends.  However, the customer from which this originates has impacted clusters with as few as 12 total nodes.  As such, please consider lowering the threshold to 8 data plane node (to trigger skipping daemonset pods.)

Reply:
If that is mainly impacting one customer at this moment, and due to the missing information reducing that value more could cause for lots of other cases (in whose it will be needed to ask for the DS pod inspection again), maybe it would be better to allow a parameter to configure that value, or even better, a parameter to disable the full inspection of DS pods, and ask that specific customer to run all their must-gathers with that specific parameter.